### PR TITLE
Remove unused condition

### LINKED ISSUE #4328: Incorrect overlapping in disabled ranges
> Clearly describe the bug

During writing types in `~/lib/assignDisabledRanges` I found problem in types [here](https://github.com/stylelint/stylelint/blob/master/lib/assignDisabledRanges.js#L120) in `ranges.end` since ranges is array of disabled ranges it does not have property `end` of type array. If rewrite it to correct syntax `!!_.last(ranges).end`, getting errors in tests with incorrect overlapping in disable ranges [here](https://github.com/stylelint/stylelint/blob/master/lib/__tests__/fixtures/disableOptions/disabled-ranges-2.css), right now `block-no-empty` is closing on line 9 because of `stylelint-disable-line` inserting new disabled range for each rule and next command `stylelint-enable` throws error since all disabled ranges have `end` already

> Which rule, if any, is the bug related to?

any

> What CSS is needed to reproduce the bug?

e.g.

```css
/* stylelint-disable block-no-empty */
a {}
/* stylelint-enable */

a {} /* stylelint-disable-line color-named */
a { color: red; } /* stylelint-disable-line block-no-empty */

/* stylelint-disable block-no-empty */
a { color: red; } /* stylelint-disable-line */
/* stylelint-enable */
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
	"rules": {
			"block-no-empty": true
		},
	}
```

> Which version of stylelint are you using?

latest

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

no

> What did you expect to happen?

I think stylelint must not close `block-no-empty` disabled range on line 9

> What actually happened (e.g. what warnings or errors did you get)?

stylelint assignDisabledRanges throwing error [here](https://github.com/stylelint/stylelint/blob/master/lib/assignDisabledRanges.js#L121)

### PR DESCRIPTION
**Pull request by hudochenkov**
> Which issue, if any, is this issue related to?

Closes #4328.

> Is there anything in the PR that needs further explanation?

I'm not sure I understood the problem correctly ¯﻿\﻿_﻿(﻿ツ﻿)﻿_﻿/﻿¯ But look like removing incorrect condition fixes types issue, and doesn't break any test.

### CODE REVIEW

**Comment by vankop** [comment]
Sorry for my bad explanation. Your code does not check `end` (right now it does not work also), but I think it is not by design. Example `disabledRanges`:
```
{
   'block-no-empty': [{start: 0, end: 0}]
}
```
for this range command `enable all` should throw error (fall in this condition) by design as I understand. Your fix ignore this behaviour

**Comment by hudochenkov** [comment]
I guess we need tests for this first. Even failing.

**Comment by hudochenkov** [comment]
@vankop do you think we can figure out the problem in the coming week? I would like to release new stylelint soon. Don't want to release another major release in case disabled ranges fix is a breaking change.

**Comment by vankop** [comment]
Actually I don't understand reason of throwing error [here](https://github.com/stylelint/stylelint/blob/master/lib/assignDisabledRanges.js#L166)... Maybe skip command is better

**Comment by hudochenkov** [comment]
We have only one test, where this condition is `true`: https://github.com/stylelint/stylelint/blob/2dfd7796f1d879d84cad1f3a922735fe5e8c6043/lib/__tests__/disableRanges.test.js#L604-L610

**Comment by vankop** [comment]
So we need to add test case:
```
/* stylelint-disable-line block-no-empty */
/* stylelint-enable */
```

and update code  to catch error in this case =)

**Comment by hudochenkov** [comment]
I've added test. This test isn't failing and without removing extra condition. In fact, we have no tests which would use run condition `_.last(ranges.end)` :) So I guess it was just a bug, that this condition was there in the first place :)

File: lib/__tests__/disableRanges.test.js [hunk 1]
```diff
@@ -238,6 +238,24 @@ it('disable single line multiple rules', () => {
 	);
 });
 
+it('disable single line one rule and re-enable all', () => {
+	return testDisableRanges(
+		'a {} /* stylelint-disable-line block-no-empty */\n/* stylelint-enable */',
+	).then((result) => {
+		expect(result.stylelint.disabledRanges).toEqual({
```
Comment (vankop) [inline lib/__tests__/disableRanges.test.js]: We must expect error here since there is nothing to enable, if I understand correct condition with `No rules have been disabled` error =)

Comment (hudochenkov) [reply lib/__tests__/disableRanges.test.js]: You're right :)

Comment (vankop) [reply lib/__tests__/disableRanges.test.js]: Hah, I found a bug in ranges assignment, I will do fix

File: lib/__tests__/needlessDisables.test.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ it('needlessDisables complex case', () => {
 				ranges: [
 					{ start: 5, end: 5, unusedRule: 'color-named' },
 					{ start: 6, end: 6, unusedRule: 'block-no-empty' },
-					{ start: 8, unusedRule: 'block-no-empty' },
+					{ start: 8, end: 10, unusedRule: 'block-no-empty' },
```
Comment (vankop) [inline lib/__tests__/needlessDisables.test.js]: test expectation was wrong..

**Comment by hudochenkov** [comment]
> -   Fixed: overlapping disabled ranges edge case ([#4497](https://github.com/stylelint/stylelint/pull/4497)).
